pkg/virt-handler/rest/lifecycle.go

```diff
@@ -240,3 +240,73 @@ func (lh *LifecycleHandler) getVMILauncherClient(request *restful.Request, respo
 
 	return vmi, client, nil
 }
+
+func (lh *LifecycleHandler) SEVFetchCertChainHandler(request *restful.Request, response *restful.Response) {
+	vmi, client, err := lh.getVMILauncherClient(request, response)
+	if err != nil {
+		return
+	}
+
+	log.Log.Object(vmi).Infof("Retrieving SEV platform info")
+
+	sevPlatformInfo, err := client.GetSEVInfo()
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to get SEV platform info")
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	response.WriteEntity(sevPlatformInfo)
+}
+
+func (lh *LifecycleHandler) SEVQueryLaunchMeasurementHandler(request *restful.Request, response *restful.Response) {
+	vmi, client, err := lh.getVMILauncherClient(request, response)
+	if err != nil {
+		return
+	}
+
+	log.Log.Object(vmi).Infof("Retreiving SEV launch measurement")
+
+	sevMeasurementInfo, err := client.GetLaunchMeasurement(vmi)
+	if err != nil {
+		log.Log.Object(vmi).Reason(err).Error("Failed to get VMI launch measurement")
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	response.WriteEntity(sevMeasurementInfo)
+}
+
+func (lh *LifecycleHandler) SEVInjectLaunchSecretHandler(request *restful.Request, response *restful.Response) {
+	vmi, client, err := lh.getVMILauncherClient(request, response)
+	if err != nil {
+		return
+	}
+
+	opts := &v1.SEVSecretOptions{}
+	if request.Request.Body != nil {
```

Comment:
Same as the comment above you can evaluate if the Body == nil and return without the else